feat(oxfmt): Enable experimental `package.json` sorting by default

leaysgur opened the pull request:
Fixes #16525, closes #16541 

- Use `sort-package-json: 0.0.2`
- Enable by default, can be disabled by `experimentalSortPackageJson: false`

Boshen commented on apps/oxfmt/src/cli/format.rs:
```diff
@@ -260,11 +263,16 @@ fn load_config_path(cwd: &Path, config_path: Option<&Path>) -> Option<PathBuf> {
 }
 
+/// Returns: (ignore_patterns, is_sort_package_json)
+type OxfmtOnlyOptions = (Vec<String>, bool);
```
Better to have a struct? We may have more options in the future. Passing a single `bool` around is not that nice as well.

leaysgur commented:
I will make some minor adjustments to improve readability.

leaysgur committed:
feat(oxfmt): Enable experimental `package.json` sorting by default (#16593)

Fixes #16525, closes #16541

- Use `sort-package-json: 0.0.2`
- Enable by default, can be disabled by `experimentalSortPackageJson: false`
